scripts/babel-6/rewrite-modules.js

```diff
@@ -0,0 +1,147 @@
+ */
+function mapModule(state, module) {
+  var moduleMap = getModuleMap(state);
+  if (moduleMap.hasOwnProperty(module)) {
+    return moduleMap[module];
+  }
+  // Jest understands the haste module system, so leave modules intact.
+  if (process.env.NODE_ENV !== 'test') {
+    var modulePrefix = getModulePrefix(state);
+    if (modulePrefix == null) {
+      modulePrefix = './';
+    }
+    return p.normalize(modulePrefix + module);
+  }
+}
+
+function getArguments(path) {
+  var args = path.get('arguments');
+  if (args && args.length) {
+    return args[0].node;
+  }
+}
+
+module.exports = function(babel) {
+
+  var t = babel.types;
+
+  /**
+   * Transforms `require('Foo')` and `require.requireActual('Foo')`.
+   */
+  function transformRequireCall(path, state) {
+    var calleePath = path.get('callee');
+    if (
+      !t.isIdentifier(calleePath.node, {name: 'require'}) &&
+      !(
+        t.isMemberExpression(calleePath.node) &&
+        t.isIdentifier(calleePath.node.object, {name: 'require'}) &&
+        t.isIdentifier(calleePath.node.property, {name: 'requireActual'})
+      )
+    ) {
+      return;
+    }
+
+    var moduleArg = getArguments(path);
+    if (moduleArg && moduleArg.type === 'StringLiteral') {
+      var module = mapModule(state, moduleArg.value);
+      if (module) {
+        path.replaceWith(
+          t.callExpression(
+            calleePath.node,
+            [t.stringLiteral(module)]
+          )
+        );
+      }
+    }
+  }
+
+  /**
+   * Transforms either individual or chained calls to `jest.dontMock('Foo')`,
+   * `jest.mock('Foo')`, and `jest.genMockFromModule('Foo')`.
+   */
+  function transformJestHelper(path, state) {
+    var calleePath = path.get('callee');
+    var args = getArguments(path);
+    if (
+      args &&
+      args.type === 'StringLiteral' &&
+      calleePath.node &&
+      (
+        t.isIdentifier(calleePath.node.property, {name: 'dontMock'}) ||
+        t.isIdentifier(calleePath.node.property, {name: 'mock'}) ||
+        t.isIdentifier(calleePath.node.property, {name: 'genMockFromModule'})
+      )
+    ) {
+      var module = mapModule(state, args.value);
+      if (module) {
+        args = t.stringLiteral(module);
+      }
+    }
+    path.replaceWith(
+      t.callExpression(
+        calleePath.node,
+        [args]
+      )
+    );
+  }
+
+  const jestIdentifier = {
+    Identifier(path) {
+      if (path.node.name === 'jest') {
+        this.isJest = true;
+      }
+    }
+  };
+
+  function transformJestCall(path, state) {
+    let params = {isJest: false};
+    path.traverse(jestIdentifier, params);
+    if (params.isJest) {
+      transformJestHelper(path, state);
+    }
+  }
+
+  return {
+    visitor: {
+      CallExpression: {
+        exit(path, state) {
+          if (path.node.seen) {
+            return;
+          }
+          transformRequireCall(path, state);
+          transformJestCall(path, state);
+          path.node.seen = true;
```

Comment:
I wonder if shouldSkip would skip all further transformations for that node, when I only want to skip the one I have defined.